tests/sessions_tests/tests.py

```diff
@@ -353,6 +354,28 @@ def test_session_load_does_not_create_record(self):
         # provided unknown key was cycled, not reused
         self.assertNotEqual(session.session_key, 'someunknownkey')
 
+    def test_ticket_21608(self):
+        # Test for #21608 - Logged out sessions are resurrected by concurrent requests.
+
+        # Create new session.
+        SESSION_KEY = 'tkniiqcv0js5reo8y0ofrywrhmwnt9xw'
+        s1 = self.backend(SESSION_KEY)
+        s1.save(must_create=True)
+
+        # Logout in an other context.
+        s2 = self.backend(SESSION_KEY)
+        s2.delete()
+
+        # Modify session in first context.
+        s1.modified = True
+        try:
+            s1.save()  # This should throw an exception as the session is deleted, not resurrect it.
+        except UpdateError:
+            pass
+
+        with self.assertRaises(Session.DoesNotExist):
+            Session.objects.get(pk=SESSION_KEY)
```

Comment:
I don't think that is possible as getting the session and updating it needs to be atomic to be safe. It could be done with a transaction for db backend that supports it, but I don't think that is a good idea.